script.js

```diff
@@ -1,4 +1,127 @@
+
+// After both players have rolled and chosen dice order, the player with the higher combined number wins.
+
+// 1st mode => 1st User runs game clicks submit => diceRoll twice => display results.
+// 2nd mode => chooses order of dice integer as strings? ex 6 and 3 would be 63. how? setting array numbers maybe.
+// 3rd mode => 2nd player runs game clicks submit => diceRoll twice => display results.
+// 4th mode => chooses order of dice integer as strings? ex 6 and 3 would be 63. how? setting array numbers maybe.
+// 5th step => compare results, if player 1 > player 2 or vice versa, the player with the highest number wins.
+
+// what do i need?
+
+// Create 1st game mode
+var GAME_STATE_DICE_ROLL = "GAME_STATE_DICE_ROLL";
+var GAME_STATE_CHOOSE_DICE_ORDER = "GAME_STATE_CHOOSE_DICE_ORDER";
+var GAME_STATE_COMPARE_SCORES = "GAME_STATE_COMPARE_SCORES";
+var gameState = GAME_STATE_DICE_ROLL;
+
+var currentPlayerRolls = [];
+
+var currentPlayer = 1;
+var allPlayerScore = [];
+
+// dice roll function
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  console.log(rollDice, "dice roll");
+  return diceNumber;
+};
+
+var rollDiceForPlayer = function () {
+  console.log("Control flow : start of rollDiceForPlayer()");
+  var counter = 0;
+  while (counter < 2) {
+    currentPlayerRolls.push(rollDice());
+    counter = counter + 1;
+  }
+  console.log("rollDiceForPlayer Changes, playerRolls: ", currentPlayerRolls);
+  return `Welcome Player ${currentPlayer}<br>You rolled ${currentPlayerRolls[0]} for Dice 1 and ${currentPlayerRolls[1]} for Dice 2.<br>Choose the order of the dice.`;
+};
+
+var getPlayerScore = function (playerInput) {
+  var playerScore;
+  // input validation
+  if (playerInput != 1 && playerInput != 2) {
+    console.log(
+      "Control flow : input validation, invalid out is not 1 and not 2"
+    );
+    return `Error Please only onput '1' or '' to choose which dice to use as the first digit.<br> Your dice rolls are :<br>Dice 1 : ${currentPlayerRolls[0]} Dice 2 : ${currentPlayerRolls[1]}.`;
+  }
+  // input == 1
+  if (playerInput == 1) {
+    console.log("Control flow : input == 1");
+    var playerScore = Number(
+      String(currentPlayerRolls[0]) + String(currentPlayerRolls[1])
+    );
+    return `Your chosen value is: ${playerScore}.`;
+  }
+  // input == 2
+  if (playerInput == 2) {
+    console.log("Control flow : input == 2");
+    var playerScore = Number(
+      String(currentPlayerRolls[0]) + String(currentPlayerRolls[1])
+    );
+    // Store playerScore rolls array
+    allPlayerScore.push(playerScore);
+    // clear current player rolls array
+    currentPlayerRolls = [];
+    return `Player ${currentPlayer}, your chosen value is ${playerScore}`;
+  }
+};
```

Comment:
i notice that we are only pushing `playerScore` to `allPlayerScore` when `playerInput` is 2, was this an intentional choice or an oversight? im thinking we need to store it regardless right?

also you probably want to put the second roll (i.e. index 1) ahead of the first roll (i.e. index 0) when the `playerInput` is 2